app/ts/components/TransactionPage/TransactionResponse.tsx

```diff
@@ -0,0 +1,74 @@
+import { useEffect } from 'preact/hooks'
+import { useTransactionResponse } from '../../store/transaction.js'
+import { TransactionResponse } from '../../types.js'
+
+type TransactionQueryProps = {
+	transactionHash: string
+	onSuccess: (response: TransactionResponse) => void
+}
+
+export const TransactionResponseQuery = (props: TransactionQueryProps) => {
+	const { transactionResponse: txn, getTransactionResponse } = useTransactionResponse()
+
+useEffect(() => {
+		getTransactionResponse(props.transactionHash)
+	}, [props.transactionHash])
+
+	switch (txn.value.state) {
```

Comment:
it may be confusing but TransactionResponse (taken from ethersjs) is a different async call from TransactionReceipt. The code block live in different files and responds on the same states so named very similarly but displays different information so can't be merged